### ISSUE TITLE
Code not showing on thank you page

### ISSUE THREAD

**Issue by ideawebservices** [enollo/woocommerce-google-trusted-stores-integration#6]
Google Trusted Stores requires the badge code to be added to every page of the site, and the order confirmation module code to be shown on the thank you page.  I installed and configured this plugin, and I see the trusted stores code when I view the source of general pages, but when I view the source of the woocommerce thank you page, there is no trusted stores code.  Neither the badge code nor the order confirmation module code is there.

**Comment by tamarazuk** [comment]
Hi @ideawebservices, really sorry for the late reply here! I can't replicate the issue here unfortunately. Could you give the new version a try (which can be downloaded [here](https://github.com/enollo/woocommerce-google-trusted-stores-integration/releases/tag/1.0.0)) and look for `gts-order` on the WooCommerce Thank You page? Also note that this code is only displayed the first time the thank you page is loaded. This is done so that the order is not tracked with Google more than once as customers may visit the receipt page multiple times.

**Comment by drnick2** [comment]
Hi Tamara, thanks for the great plugin, I just have a couple of small issues with version 1.0.0, first I am getting a  java warning from google on a missing java code where they want a ID it is"google_base_offer_id" field of the google Trusted Stores. I was trying to add it but have not been able to locate it. Second issue on checkout I am not getting any  google order confirmation for the opt in which is necessary to complete googles test of our site, any help would be appreciated.

**Comment by tamarazuk** [comment]
Hi @drnick2, the `google_base_offer_id` is only shown on product pages - are you seeing this warning on any product pages? If you only see the warning on non-product pages, then it is safe to ignore :smile_cat: 

> Second issue on checkout I am not getting any google order confirmation for the opt in which is necessary to complete googles test of our site, any help would be appreciated.

I am not sure if I understand the second issue. Could you elaborate a bit on the issue or provide the errors Google is giving? Thanks!

**Comment by drnick2** [comment]
Hi Tamara,

 

Thanks for your reply, I only saw that warning on the main page which is a non product page, thanks for that info, now for tne 2nd issue I came across.

 

I C/P below where Google shows a  couple of steps needed for the second part of their test.

 

Test order

Complete test orders both as a guest and with a signed-in account, with a real email address. On the order confirmation page, click “Yes” to opt in for purchase protection. If you do not see the opt-in on the order confirmation page, this means the order confirmation module code is not detected by Google. Review any error messages on the yellow bar. Learn more

 

Purchase protection email

Open the purchase protection email from Google, and click through to your Google Trusted Stores customer account. Review your order and ensure that the estimated ship date and order details are accurate, to verify that you are populating correct values through your code.

 

Where I have it highlighted in the first paragraph is what I am not seeing when I place an order. It just goes to the completed order page showing the order has been paid and completed.  I guess without that I cannot take care of the 2nd part of the test I highlighted.

 

Thanks for your time and help.

 

Best,

 

 Nick

**Comment by nicksarr1** [comment]
Running into the same issue here as @drnick2

**Comment by nicksarr1** [comment]
The google protection appears only after the transaction is completed. It appears on the Thank you page after the order.

**Comment by elituchy** [comment]
I'm also having issue on the Order confirmation module (see screenshot from Google Test page).
![google trusted stores clipular](https://cloud.githubusercontent.com/assets/7242085/9117076/f36ecdf4-3c35-11e5-9a52-f501efa1b7d5.png)

**Comment by nicksarr1** [comment]
Elituchy, 
Did you fully complete a purchase on each browser? I had the same issue, but I wasn't fully completing an order. I had to actually make a test product for .01 and purchase it. On the Thankyou page (after you enter your card info) There is a pop up that appears from google. You must accept to their warranty option as that is apart of the verification. 

After completing and a purchase and accepting the google terms on the thank you page. The Order Confirmation Module will change to Passing. 

The trusted stores badge I found will only show passed on pages with products. So there will always be a warning, but as long as it says passed with warnings you're fine.

**Comment by drnick2** [comment]
I did a complete purchase on my browser with a credit card, but I received no popup and thus still have not been able to pass the Google confirmation requirement, Did anybody modify or add script to there web pages or server or did you strictly used the plugin?  Any help is appreciated,thanks.

**Comment by elituchy** [comment]
Nick, we get a handful of daily orders from this website so I'm certain at least one person has ordered.  I haven't tried it personally.  BUT... my screenshot from a Google tool, so it would be able to check the page for the code even without an order being placed. Correct?

**Comment by elituchy** [comment]
Update!  I tested it and it DOES work.  I see the popup on the confirmation page.

One last issue... Google is saying we need to "Provide the Google Shopping Item ID in the “google_base_offer_id” field of the Google Trusted Stores JavaScript code."  Should the plugin do this automatically or is it done manually?

**Comment by nicksarr1** [comment]
Awesome! Yeah it took me a bit to get it fully working. I'm pretty sure the plugin takes care of the rest. I just installed and the submitted the site to google. Didn't have to do anything special.

**Comment by drnick2** [comment]
Unfortunately I did not get a popup this morning when I completed a transaction. The plugin should handle everything, but not for me presently. Did you use the checkbox in the advanced option section to use the older Google Trusted Stores JavaScript?

**Comment by tamarazuk** [comment]
> One last issue... Google is saying we need to "Provide the Google Shopping Item ID in the “google_base_offer_id” field of the Google Trusted Stores JavaScript code." Should the plugin do this automatically or is it done manually?

@elituchy The plugin does do this automatically, but only if you are viewing a product page. Let me know if it's not working properly on the product page, or if you expect it to be on other pages (although I don't see how as it should have the product ID as it's value).

> Unfortunately I did not get a popup this morning when I completed a transaction. The plugin should handle everything, but not for me presently. Did you use the checkbox in the advanced option section to use the older Google Trusted Stores JavaScript?

@drnick2 You should only need to use that checkbox if your Google Trusted Stores account is a bit older and tracking is not working at all. If you are having intermittent problems, there might be another issue. 

To all commenting and helping out, thank you very much! I try my best to maintain this plugin, but unfortunately, Google Trusted Stores is not available in my country yet and I don't have access to an account anymore. If anyone would like to get involved by helping test through there account please let me know. It would make troubleshooting a lot easier :)

**Comment by drnick2** [comment]
I might like to volunteer, what exactly would you need access to.

**Comment by tamarazuk** [comment]
@drnick2 i would need to be added as an admin on one of your stores. Is it possible to have a 'dummy' account within Google Trusted Stores? I understand it takes some time to get verified, but I am not sure if they have any restrictions on it being a sandbox site without any real data.

**Comment by princebhalani143** [comment]
Hi all,

From my experience it's not taking time to get approve from google trusted store, You can set up this all in 15 min if you have every detail
